ssg/__init__.py

```diff
@@ -0,0 +1,14 @@
+import ssg._checks as checks
+import ssg._constants as constants
+import ssg._id_translate as id_translate
+import ssg._jinja as jinja
+import ssg._map_product as map_product
+import ssg._parse_oval as parse_oval
+import ssg._stig as stig
+import ssg._utils as utils
+import ssg._xccdf as xccdf
+import ssg._xml as xml
+import ssg._yaml as yaml
+
+# Compatibility shims
+import ssg._shims as shims
```

Comment:
Having an underscore `_` seems weird to me.... I would expect to not have an `underscore` at all.

Reply:
I am not 100% sure but I think this is to prevent name clashes. For example you need `_yaml` to avoid `import yaml` importing your own module inside your own module...

This `import _abc as abc` just renames it in the ssg package so that we have nice package names when we are using the module.

Reply:
Yeah, @mpreisler is correct here. There's some workaround with absolute vs. relative paths that you can do, but both `xml` and `yaml` conflict, and naming `_xml` instead of `xml` ends up being shorter. I opted to make everything consistent after that, so that's why, e.g., `_shims` begins with an underscore.

When loading from outside, `import ssg` will import everything with expected names (so you can use `ssg.xml.parse_xml_file(...)`), but if you want to import specific functions, you have to use the underscore (e.g., `from ssg._xml import parse_xml_file` or `import ssg._stig as stig`).

Reply:
Python3 has relative imports that should prevent shadowing of imports. I am not against underscores, though.
What should be changed are scripts that define functions and that have dash in their names, s.a. `combine-ovals.py`. Those files can't be imported normally, and therefore can't be unit-tested.

Reply:
@matejak they can be imported normally. check out `__import__`

Reply:
@mpreisler I know that it can be done, by "normal imports" I meant import statements. It is not accurate that those files couldn't be unit tested. However, I think that files that define functions that are testable, should have import-friendly names.

Reply:
@matejak I hope to address this in a second PR after this lands. I want to take everything in `build-scripts` and `utils` and move functionality into the `ssg` module such that it can be better unit tested... That way `build-scripts` and `utils` are just wrappers for calling (main methods). When I move them into `build-scripts`, I'll clean up their names.

Reply:
@cipherboy I like the idea not to do everything in this PR.
Just to name an example, the `generate_contributors.py` has a `main` function. That one could be split into multiple functions, but it doesn't make economical sense to unit-test any of them - they just generate text. The rest of the functionality could be imported from a separate package and separate module with underscores in their names.
The script itself can go back to being `generate-contributors.py`.

Reply:
Sure, `generate-contributors` was just an example.

Reply:
@matejak Is https://github.com/OpenSCAP/scap-security-guide/pull/2967/commits/1c0d49006f68f4b92831335824a78901efba8501 from #2967 what you're thinking?

Reply:
(Note that I'm going to turn all the _build_*.py scripts into a proper submodule after this, but it is an example of an issue from my other comment [here](https://github.com/OpenSCAP/scap-security-guide/pull/2949#issuecomment-397594152).